webapp/components/expression/nodes/identifier.js

```diff
@@ -1,16 +1,41 @@
 import React from 'react'
 import * as R from 'ramda'
+import PropTypes from 'prop-types'
 
-import Dropdown from '@webapp/components/form/Dropdown'
+import * as A from '@core/arena'
 
-const Identifier = ({ node, variables, onChange }) => (
-  <Dropdown
-    items={variables}
-    selection={R.find(R.propEq('value', node.name), variables)}
-    itemLabel="label"
-    itemKey="value"
-    onChange={item => onChange(R.assoc('name', R.propOr('', 'value', item), node))}
-  />
-)
+import Select from '@webapp/components/form/Select'
+
+const findValue = ({ variables, node }) => {
+  let value = null
+  value = R.find(R.propEq('value', node.name), variables)
+  if (!A.isEmpty(value)) return value
+  const _flatVariables = variables.reduce((flat, group) => [...(flat || []), ...(group.options || [])], [])
```

Comment:
can flat be null? ...(flat || [])  probably not.
also, give meaningful names. what is flat? variablesAggregator?

Reply:
Changed!